cmd/sonobuoy/app/e2e.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"compress/gzip"
+	"fmt"
+	"os"
+
+	"github.com/heptio/sonobuoy/pkg/client"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	show        string
+	rerunFailed bool
+)
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "e2e archive.tar.gz",
+		Short: "Inspect e2e test results. Optionally rerun failed tests",
+		Run:   e2es,
+		Args:  cobra.ExactArgs(1),
+	}
+	AddGenFlags(&runopts.GenConfig, cmd)
+	AddModeFlag(&runFlags.mode, cmd)
+	AddSonobuoyConfigFlag(&runFlags.sonobuoyConfig, cmd)
```

Comment:
could you make another local struct for this? Using runFlags is fine by me, but I think show and rerun failed should go in there too (since the namespace is still shared)